arviz/data/io_pyro.py

```diff
@@ -1,119 +1,131 @@
-"""pyro-specific conversion code."""
+"""Pyro-specific conversion code."""
+import logging
+import numpy as np
+
 from .inference_data import InferenceData
-from .base import dict_to_dataset
+from .base import requires, dict_to_dataset
 from .. import utils
 
-
-def _get_var_names(posterior):
-    """Extract latent and observed variable names from pyro.MCMC.
-
-    Parameters
-    ----------
-    posterior : pyro.MCMC
-        Fitted MCMC object from Pyro
-
-    Returns
-    -------
-    list[str], list[str]
-    observed and latent variable names from the MCMC trace.
-    """
-    sample_point = posterior.exec_traces[0]
-    nodes = [node for node in sample_point.nodes.values() if node["type"] == "sample"]
-    observed = [node["name"] for node in nodes if node["is_observed"]]
-    latent = [node["name"] for node in nodes if not node["is_observed"]]
-    return observed, latent
+_log = logging.getLogger(__name__)
 
 
 class PyroConverter:
     """Encapsulate Pyro specific logic."""
 
-    def __init__(self, *, posterior, coords=None, dims=None):
-        """Convert pyro data into an InferenceData object.
+    nchains = None  # type: int
+    ndraws = None  # type: int
+
+    def __init__(
+        self, *, posterior=None, prior=None, posterior_predictive=None, coords=None, dims=None
```

Comment:
I would allow `posterior` to be a positional argument. The * shoild go between posterior and prior

Reply:
I think so too. But this is a convention of `io_pystan` and `io_pymc3`. This is not important I guess.

Reply:
Yeah, for the class init this is ok (no one really use that class directly)